sklearn/decomposition/_dict_learning.py

```diff
@@ -1006,6 +1006,21 @@ class SparseCoder(SparseCodingMixin, BaseEstimator):
     components_ : array, [n_components, n_features]
         The unchanged dictionary atoms
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.decomposition import SparseCoder
+    >>> X = np.array([[1, 1, 1, 1]])
+    >>> dictionary = np.array([
+    ...             [10, 10, 10, 10],
+    ...             [5, 5, 5, 5]
+    ... ])
+    >>> coder = SparseCoder(dictionary=dictionary,
+    ...                     transform_algorithm='threshold',
+    ...                     transform_alpha=5)
+    >>> coder.transform(X)
+    array([[35, 15]])
```

Comment:
I think it would be more interesting to build an example that shows that the code (the output of `transform`) can be sparse, that is with a majority of zeros.

Perhaps this examples could be reworked to use a dictionarry of size `(n_components, n_features) == (5, 3)` and `X.shape (n_samples, n_features)` of size `(2, 3)` with code vectors that have at least 3 zeros out of 5 code space dimensions.

Reply:
Hi I didn't understand this bit

Reply:
Something like the following:

```python
>>> import numpy as np 
>>> from sklearn.decomposition import SparseCoder 
>>> X = np.array([[-1, -1, -1], [0, 0.5, 0]]) 
>>> dictionary = np.array([ 
...             [0, 1, 0], 
...             [-1, -1, 2], 
...             [1, 1, 1], 
... ]).astype(np.float64) 
>>> coder = SparseCoder(dictionary=dictionary, 
...                     transform_algorithm='omp') 
>>> coder.transform(X)
array([[ 0. ,  0. , -1. ],
       [ 0.5,  0. ,  0. ]])
```

Here you see that the input data naturally decompose into a sparse signal because the input vectors align simply with the dictionary components.

I used omp (Orthogonal Matching Pursuit) because I could not figure how to use the "threshold" method :)

Reply:
I used only 5 dictionary elements, it could be a bit more interesting with 5 dict components (sparse over complete representation) to get one than more non-zero element in encoded vectors while still having a majority of zeros.

Reply:
Here is an alternative example that's interesting with 5 dict components:

```
>>> import numpy as np  
>>> from sklearn.decomposition import SparseCoder  
>>> X = np.array([[-1, -1, -1], [0, 0, 3]])  
>>> dictionary = np.array([  
...             [0, 1, 0],  
...             [-1, -1, 2],  
...             [1, 1, 1],  
...             [0, 1, 1], 
...             [0, 2, 1], 
... ]).astype(np.float64)  
>>> coder = SparseCoder(dictionary=dictionary,  
...                     transform_algorithm='lasso_lars',
...                     transform_alpha=1e-10) 
>>> coder.transform(X)                                                                                                                                                 
array([[ 0.,  0., -1.,  0.,  0.],
       [ 0.,  1.,  1.,  0.,  0.]])
```